pandas/core/ops.py

```diff
@@ -1151,7 +1151,7 @@ def dispatch_to_extension_op(op, left, right):
                 new_right = [new_right]
             new_right = list(new_right)
         elif is_extension_array_dtype(right) and type(left) != type(right):
-            new_right = list(new_right)
+            new_right = list(right)  # TODO: was this intended?
```

Comment:
Encountered the same error while working on EA-Setarray - gets UnBoundLocalError. Actually, I'd transition to something like:
```
elif is_extension_array_dtype(right) and type(left) != type(right):
    new_right = new_right.astype(left.dtype).values
elif is_extension_array_dtype(right):
    new_right = right.values
else:
    new_right = right
```
otherwise the errors become cryptic (e.g. applying `op(EA, EA)` giving an error about not being compatible with `list` - which the user never sees in the objects he inputs).

Reply:
yes, this is in a separate issue I think

Reply:
Separate PR?  I'll be happy to make it (especially if there's already a test)

Reply:
Looks like https://github.com/pandas-dev/pandas/pull/22479/files has it